### PR TITLE
FortranCon 2021 stdlib abstract

### PR DESCRIPTION
**Pull request by nshaffer**
Hello all! This is a first draft of an abstract for a talk on the past year's developments in stdlib, to be submitted **7/31/2021**. The tentative author list includes all GitHub users who made at least one [contribution](https://github.com/fortran-lang/stdlib/graphs/contributors) since last July:

@ghbrown @certik @wclodius2 @rscohn2 @milancurcic @LKedward @awvwgk @gareth-nx @Aman-Godara @Jim-215-Fisher @ChetanKarwa @arjenmarkus @14NGiestas @wyphan @ivan-pi @sakamoti @hsnyder @jvdp1 @ejovo13 @brocolis 

What you can do for me:

1. Confirm here that you would like to be an author. I am treating authorship as **opt-in**, meaning that if I do not hear from you before the abstract submission deadline (7/31/2021), your name will not be included. However, I plan to have an acknowledgements slide that will list all stdlib contributors to date.
2. Check your affiliation, or provide one if it is not listed. If you prefer not to have an affiliation listed, just comment below or email me at nrshaffer@protonmail.com
3. If you think I have omitted any contributors from the past year, please comment below!

The abstract I have written is very generic and short. My intention is to leave space for you all to suggest additions. For instance, if there is a particular new module (or modules) that you think should be highlighted in the abstract, please bring that up! Likewise, I have not talked about the build system improvements in any detail. This is because I am not very familiar with those changes. If someone who was involved with that can comment on that, that would be a big help as well.

### CODE REVIEW

File: FortranCon2021-stdlib/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+[Aman Godara](https://github.com/Aman-Godara),
+[Jing ???](https://github.com/Jim-215-Fisher),
+[Chetan Karwa](https://github.com/chetankarwa),
+[Arjen Markus](https://github.com/arjenmarkus),
+[Ian Giestas Pauli](https://github.com/14NGiestas) (São Paolo University),
+[Wileam Phan](https://github.com/wyphan) (University of Tennessee, Knoxville), 
+[Ivan Pribec](https://github.com/ivan-pi), (Technical University of Munich),
+[Yuichiro Sakamoto](https://github.com/sakamoti),
+[Harris Snyder](https://github.com/hsnyder), 
+[Jeremie Vandenplas](https://github.com/jvdp1) (Wageningen University & Research),
```
Comment (jvdp1) [inline FortranCon2021-stdlib/README.md:19]: Thank you for starting this. 
Good for me. I'll come back on the abstract ASAP.

File: FortranCon2021-stdlib/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# FortranCon2020-stdlib
```
Comment (aman-godara) [inline FortranCon2021-stdlib/README.md]: 2021?

Comment (ivan-pi) [reply FortranCon2021-stdlib/README.md]:
```suggestion
# FortranCon2021-stdlib
```

File: FortranCon2021-stdlib/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+**Authors**:
+[Nathaniel Shaffer](https://github.com/nshaffer) (University of Rochester), 
+[Gabriel Brown](https://github.com/ghbrown) (University of Illinois Urbana-Champagne), 
+[Ondřej Čertik](https://github.com/certik) (Los Alamos National Laboratory),
+[William Clodius](https://githiub.com/wclodius2),
+[Robert Cohn](https://github.com/rscohn2) (Intel Corporation)
+[Milan Curcic](https://github.com/milancuric) (University of Miami),
+[Laurence Kedward](https://github.com/lkedward) (University of Bristol),
+[Sebastian Ehlert](https://github.com/awvwgk) (University of Bonn),
+[gareth-nx ???](https://github.com/gareth-nx),
```
Comment (gareth-nx) [inline FortranCon2021-stdlib/README.md]: [gareth-nx Gareth Davies](https://github.com/gareth-nx), 

For administrative reasons it's simpler for me to have no affiliation (also this is unrelated to my day job).

File: FortranCon2021-stdlib/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+**Track**: Junior presentations (15 + 5 minutes discussion)
+
+**Presenter**: [Nathaniel Shaffer](https://github.com/nshaffer)
+
+**Title**: Fortran Standard Library
+
+**Abstract**:
+The Fortran standard library is now well into its second year of development.
+In this past year, stdlib has grown not only in its functionality but in its contributor base, with most of the new development coming from new contributors, including two Google Summer of Code students.
```
Comment (milancurcic) [inline FortranCon2021-stdlib/README.md]: What do you think about adding a sentence here about new functionality?

```suggestion
In this past year, stdlib has grown not only in its functionality but in its contributor base, with most of the new development coming from new contributors, including two Google Summer of Code students.
Stdlib provides new functionality for working with bit sets, logging, quadrature, sorting, strings, and more.
```

File: FortranCon2021-stdlib/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+[Carlos Une](https://github.com/brocolis)
+
+
+**To be acknowledged in the presentation (following comments):**
+
+**Track**: Junior presentations (15 + 5 minutes discussion)
+
+**Presenter**: [Nathaniel Shaffer](https://github.com/nshaffer)
+
+**Title**: Fortran Standard Library
```
Comment (milancurcic) [inline FortranCon2021-stdlib/README.md]: What do you think about making title a bit more specific, for example:

```suggestion
**Title**: What's new in the Fortran Standard Library?
```

or

```suggestion
**Title**: Second year of the Fortran Standard Library
```

**Review by milancurcic — APPROVED** [review]
Looks good, thanks for starting it! I added a few comments. Yes, I'm opting in as co-author.

File: FortranCon2021-stdlib/README.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# FortranCon2021-stdlib
+
+**Authors**:
+[Nathaniel Shaffer](https://github.com/nshaffer) (University of Rochester), 
+[Gabriel Brown](https://github.com/ghbrown) (University of Illinois Urbana-Champagne), 
+[Ondřej Čertik](https://github.com/certik) (Los Alamos National Laboratory),
+[William Clodius](https://githiub.com/wclodius2),
+[Robert Cohn](https://github.com/rscohn2) (Intel Corporation)
```
Comment (rscohn2) [inline FortranCon2021-stdlib/README.md]:
```suggestion
```

I only contributed a minor fix to a CI file. You can omit me. Thanks.

File: FortranCon2021-stdlib/README.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+[Milan Curcic](https://github.com/milancuric) (University of Miami),
+[Laurence Kedward](https://github.com/lkedward) (University of Bristol),
+[Sebastian Ehlert](https://github.com/awvwgk) (University of Bonn),
+[gareth-nx ???](https://github.com/gareth-nx),
+[Aman Godara](https://github.com/Aman-Godara),
+[Jing ???](https://github.com/Jim-215-Fisher),
+[Chetan Karwa](https://github.com/chetankarwa),
+[Arjen Markus](https://github.com/arjenmarkus),
+[Ian Giestas Pauli](https://github.com/14NGiestas) (São Paolo University),
```
Comment (14NGiestas) [inline FortranCon2021-stdlib/README.md]:
```suggestion
[Ian Giestas Pauli](https://github.com/14NGiestas) (University of São Paulo),
```
I also made just a quick CI fix, so feel free to omit it. I've tried to contribute with discussions and some code review early on but not so much.

File: FortranCon2021-stdlib/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+**To be acknowledged in the presentation (following comments):**
+
+**Track**: Junior presentations (15 + 5 minutes discussion)
+
+**Presenter**: [Nathaniel Shaffer](https://github.com/nshaffer)
+
+**Title**: Fortran Standard Library
+
+**Abstract**:
+The Fortran standard library is now well into its second year of development.
```
Comment (jvdp1) [inline FortranCon2021-stdlib/README.md]:
```suggestion
The Fortran standard library (stdlib) is now well into its second year of development.
```

File: FortranCon2021-stdlib/README.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+In this past year, stdlib has grown not only in its functionality but in its contributor base, with most of the new development coming from new contributors, including two Google Summer of Code students. 
+In addition to new capabilities in stdlib itself, there have also been several technical improvements to the documentation, workflow, and build system.
```
Comment (jvdp1) [inline FortranCon2021-stdlib/README.md]:
```suggestion
For example, in addition to new capabilities in stdlib itself, there have also been several technical improvements to the documentation, and build system.
The full API documentation is generated by the FORD tool, and can be found online.
Furthermore, the Github 's Continuous Integration pipeline is used to test all procedures derived types and modules at all pull requests, to ensure cross-platform support.
```

**Review by jvdp1 — COMMENTED** [review]
Thank you. Here are some suggestions.

File: FortranCon2021-stdlib/README.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+[Milan Curcic](https://github.com/milancuric) (University of Miami),
+[Laurence Kedward](https://github.com/lkedward) (University of Bristol),
+[Sebastian Ehlert](https://github.com/awvwgk) (University of Bonn),
+[gareth-nx ???](https://github.com/gareth-nx),
+[Aman Godara](https://github.com/Aman-Godara),
+[Jing ???](https://github.com/Jim-215-Fisher),
+[Chetan Karwa](https://github.com/chetankarwa),
+[Arjen Markus](https://github.com/arjenmarkus),
+[Ian Giestas Pauli](https://github.com/14NGiestas) (São Paolo University),
+[Wileam Phan](https://github.com/wyphan) (University of Tennessee, Knoxville),
```
Comment (wyphan) [inline FortranCon2021-stdlib/README.md]: I only contributed a minor fix for CMake, so please feel free to omit my name, for now. Looking forward to contribute more in the future, though!
```suggestion
```

File: FortranCon2021-stdlib/README.md (hunk 3 above)
Comment (nshaffer) [reply FortranCon2021-stdlib/README.md]:
```suggestion
[Gareth Davies](https://github.com/gareth-nx),
```

**Review by wyphan — CHANGES_REQUESTED** [review]

File: FortranCon2021-stdlib/README.md (hunk 9 above)
Comment (nshaffer) [reply FortranCon2021-stdlib/README.md]: Thanks, Jeremie. I'll adjust the phrasing to my taste later, but these are nice suggestions.

File: FortranCon2021-stdlib/README.md (hunk 5 above)
Comment (nshaffer) [reply FortranCon2021-stdlib/README.md]: I like that "What's new..." title. Thanks for the suggestion.

**Comment by wclodius2** [comment]
I believe I have made significant enough contribution that I deserve to be a co-author. I am retired so I should have no affiliation.

File: FortranCon2021-stdlib/README.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
+# FortranCon2021-stdlib
+
+**Authors**:
+[Nathaniel Shaffer](https://github.com/nshaffer) (University of Rochester), 
+[Gabriel Brown](https://github.com/ghbrown) (University of Illinois Urbana-Champagne),
```
Comment (ghbrown) [inline FortranCon2021-stdlib/README.md]:
```suggestion
[Gabriel Brown](https://github.com/ghbrown) (University of Illinois Urbana-Champaign), 
```
I am also opting in. Thanks!

File: FortranCon2021-stdlib/README.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+**Track**: Junior presentations (15 + 5 minutes discussion)
+
+**Presenter**: [Nathaniel Shaffer](https://github.com/nshaffer)
+
+**Title**: What's new in the Fortran Standard Library?
+
+**Abstract**:
+The Fortran standard library (stdlib) is now well into its second year of development.
+In this past year, stdlib has grown not only in its functionality but in its contributor base, with most of the new development coming from new contributors, including two Google Summer of Code students.
+Stdlib provides new functionality for working with bit sets, logging, quadrature, sorting, strings, and more.
```
Comment (ivan-pi) [inline FortranCon2021-stdlib/README.md:34]:
```suggestion
Stdlib provides new functionality for working with bitsets, logging, quadrature, sorting, strings, and more.
```

**Review by ivan-pi — APPROVED** [review]
I would also like to opt-in.

**Comment by awvwgk** [comment]
Thanks for starting this. I'm happy to help with the talk as author.

File: FortranCon2021-stdlib/README.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+[Chetan Karwa](https://github.com/chetankarwa),
+[Arjen Markus](https://github.com/arjenmarkus),
+[Ian Giestas Pauli](https://github.com/14NGiestas) (São Paolo University),
+[Wileam Phan](https://github.com/wyphan) (University of Tennessee, Knoxville), 
+[Ivan Pribec](https://github.com/ivan-pi), (Technical University of Munich),
+[Yuichiro Sakamoto](https://github.com/sakamoti),
+[Harris Snyder](https://github.com/hsnyder), 
+[Jeremie Vandenplas](https://github.com/jvdp1) (Wageningen University & Research),
+[Evan Voyles](https://github.com/ejovo13) (Kalamazoo College),
+[Carlos Une](https://github.com/brocolis)
```
Comment (ghost) [inline FortranCon2021-stdlib/README.md]:
```suggestion
```
I opt-out. Thanks.

File: FortranCon2021-stdlib/README.md [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+[Sebastian Ehlert](https://github.com/awvwgk) (University of Bonn),
+[Gareth Davies](https://github.com/gareth-nx),
+[Aman Godara](https://github.com/Aman-Godara),
+[Jing ???](https://github.com/Jim-215-Fisher),
+[Chetan Karwa](https://github.com/chetankarwa),
+[Arjen Markus](https://github.com/arjenmarkus),
+[Ian Giestas Pauli](https://github.com/14NGiestas) (São Paolo University),
+[Wileam Phan](https://github.com/wyphan) (University of Tennessee, Knoxville), 
+[Ivan Pribec](https://github.com/ivan-pi), (Technical University of Munich),
+[Yuichiro Sakamoto](https://github.com/sakamoti),
```
Comment (sakamoti) [inline FortranCon2021-stdlib/README.md]: I only modified some code to work with a particular compiler, so feel free to omit me. 
Thank you so much.
```suggestion
```

File: FortranCon2021-stdlib/README.md [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+
+**Presenter**: [Nathaniel Shaffer](https://github.com/nshaffer)
+
+**Title**: What's new in the Fortran Standard Library?
+
+**Abstract**:
+The Fortran standard library (stdlib) is now well into its second year of development.
+In this past year, stdlib has grown not only in its functionality but in its contributor base, with most of the new development coming from new contributors, including two Google Summer of Code students.
+Stdlib provides new functionality for working with bit sets, logging, quadrature, sorting, strings, and more.
+For example, in addition to new capabilities in stdlib itself, there have also been several technical improvements to the documentation, and build system.
```
Comment (certik) [inline FortranCon2021-stdlib/README.md]:
```suggestion
For example, in addition to new capabilities in stdlib itself, there have also seen several technical improvements to the documentation, and build system.
```

File: FortranCon2021-stdlib/README.md [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+The full API documentation is generated by the FORD tool, and can be found online.
+Furthermore, the Github 's Continuous Integration pipeline is used to test all procedures derived types and modules at all pull requests, to ensure cross-platform support.
```
Comment (certik) [inline FortranCon2021-stdlib/README.md]:
```suggestion
Furthermore, the Github's Continuous Integration pipeline is used to test all procedures derived types and modules at all pull requests, to ensure cross-platform support.
```

**Review by certik — APPROVED** [review]
Thank you for adding me as an author. Let me know how I can help.

File: FortranCon2021-stdlib/README.md (hunk 15 above)
Comment (certik) [reply FortranCon2021-stdlib/README.md]: My bad, the original is correct.

File: FortranCon2021-stdlib/README.md [hunk 17]
```diff
@@ -0,0 +1,43 @@
+# FortranCon2021-stdlib
+
+**Authors**:
+[Nathaniel Shaffer](https://github.com/nshaffer) (University of Rochester), 
+[Gabriel Brown](https://github.com/ghbrown) (University of Illinois Urbana-Champaign), 
+[Ondřej Čertik](https://github.com/certik) (Los Alamos National Laboratory),
+[William Clodius](https://githiub.com/wclodius2),
+[Robert Cohn](https://github.com/rscohn2) (Intel Corporation)
+[Milan Curcic](https://github.com/milancuric) (University of Miami),
```
Comment (aman-godara) [inline FortranCon2021-stdlib/README.md:8]:
```suggestion
[Milan Curcic](https://github.com/milancurcic) (University of Miami),
```
@milancurcic

**Comment by ejovo13** [comment]
This is kinda random but in the talk you should definitely mention how the (subjective?) popularity of Fortran has grown at an incredible rate since 2020, at least according to the TIOBE index. I've been watching it over the last six months or so as it first entered the top 20 around May, and now It's climbed all the way up to 14! To put that into perspective, one year ago it was ranked 50th...

screenshot

The rise in rank is clearly a reflection of the effort that the community has put in to reanimate this quirky language.

**Comment by awvwgk** [comment]
Another interesting set of statistics can be found at codecov's [State of Open Source Code Coverage](https://about.codecov.io/resource/2020-state-of-open-source-code-coverage/) where a significant growth in Fortran based projects is visible.

I was planning to get some statistics for the fpm presentation together on the adoption of fpm in Fortran projects (projects with `fpm.toml`, developers with fpm enabled projects, ...).

File: FortranCon2021-stdlib/README.md (hunk 17 above)
Comment (nshaffer) [reply FortranCon2021-stdlib/README.md:8]: Good catch!

**Comment by nshaffer** [comment]
Thanks all for the feedback! I have just pushed a second draft. I intend to submit this Friday evening, Eastern Standard Time (UTC-05:00). If you have any final revisions or additions you'd like to suggest, speak now or forever hold your peace!

There are still a handful of contributors I am hoping to hear from.
@Jim-215-Fisher 
@ChetanKarwa 
@arjenmarkus 

I will also try getting in touch via the Discourse.

**Comment by Jim-215-Fisher** [comment]
Thank you for including my name in the list. My suggestion is that we might give some focus on math in stdlib. When people think of Fortran, people think numerical calculation. If we have more readily available procedures in stdlib, we can attract more people to Fortran.

**Comment by nshaffer** [comment]
Thanks for that suggestion, @Jim-215-Fisher. I will definitely mention mathematical functions in the talk when I speak about the scope of stdlib and its goals. There's been some progress in the past year, but (as you've experienced) non-trivial mathematical procedures are difficult to review and adopt into stdlib. I plan to mention this in the talk as well.

What name/affiliation would you like me to use for you? If you prefer to keep your real name private, I can list an alias.

**Review by LKedward — APPROVED** [review]
Thanks @nshaffer, this looks good. And yes I'm opting in as a co-author.

**Comment by scivision** [comment]
I was omitted: @scivision

**Comment by hsnyder** [comment]
I'd like to opt-in as well - thanks for listing me as an author. My affiliation is "Structura Biotechology Inc". I'm happy to help more if directed.

**Comment by nshaffer** [comment]
Thanks all for your feedback! I've merged this into master. I'm hitting some strange error with the abstract submission system. I've contacted the FortranCon organizers and will hopefully have it resolved in time to submit tomorrow.

Edit: Not sure what I did differently this time, but I got it in!
